src/core/qgsexpression.cpp

```diff
@@ -2921,6 +2921,25 @@ static QVariant fcnProject( const QVariantList &values, const QgsExpressionConte
   return QVariant::fromValue( QgsGeometry( new QgsPointV2( newPoint ) ) );
 }
 
+static QVariant fcnInclination( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom1 = getGeometry( values.at( 0 ), parent );
+  QgsGeometry fGeom2 = getGeometry( values.at( 1 ), parent );
+
+  const QgsPointV2 *pt1 = static_cast<const QgsPointV2 *>( fGeom1.geometry() );
```

Comment:
better make these dynamic_casts, or move the wkb type check before this static_cast